### PR TITLE
Some updates to oslc4j-client including package rename, and HTTP error processing

### PR DESCRIPTION
**Pull request by jamsden**
I renamed the packages to be more consistent with oslc4j-core, and to avoid confusion with the old oslc-java-client. 

Added some HTTP error processing in the OslcClient discovery methods where the OslcClient is acting like a client itself. 

Added some error processing in the sample applications to fail if they aren't able to do discovery for any reason (including failed authentication).

### CODE REVIEW

File: oslc4j-client/pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <name>OSLC Java Client</name>
     <description>Eclipse Lyo OSLC Java client based on OSLC4J and JAX-RS 2.0.</description>
     <scm>
-        <url>http://git.eclipse.org/c/lyo/org.eclipse.lyo.client.git</url>
+        <url>http://git.eclipse.org/c/lyo/org.eclipse.lyo.oslc4j.client.git</url>
```
Comment (berezovskyi) [inline oslc4j-client/pom.xml]: wrong URL, just remove the scm tag

File: oslc4j-client/pom.xml [hunk 2]
```diff
@@ -98,7 +98,7 @@
        <dependency>
             <groupId>org.glassfish.jersey.core</groupId>
             <artifactId>jersey-client</artifactId>
-            <version>2.27</version>
+            <version>2.28</version>
```
Comment (berezovskyi) [inline oslc4j-client/pom.xml]: CQ?

File: oslc4j-client-samples/src/main/java/org/eclipse/lyo/client/oslc/samples/RMSample.java [hunk 3]
```diff
@@ -87,6 +88,11 @@ public static void main(String[] args) throws ParseException {
 
 			//STEP 2: Find the OSLC Service Provider for the service provider we want to work with
 			String serviceProviderUrl = client.lookupServiceProviderUrl(catalogUrl, providerTitle);
+			if (serviceProviderUrl == null) {
+				System.err.println("Cannot perofrm OSLC discover, exiting...");
```
Comment (berezovskyi) [inline oslc4j-client-samples/src/main/java/org/eclipse/lyo/client/oslc/samples/RMSample.java]: typos

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java [hunk 4]
```diff
@@ -441,6 +441,10 @@ public String lookupServiceProviderUrl(final String catalogUrl, final String ser
 	{
 		String retval = null;
 		Response response = getResource(catalogUrl,OSLCConstants.CT_RDF);
+		if (response.getStatus() != HttpStatus.SC_OK) {
+			logger.error("Cannot read "+catalogUrl+" status: "+response.getStatus());
+			return null;
```
Comment (berezovskyi) [inline oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java]: Why return null?

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java [hunk 5]
```diff
@@ -441,6 +441,10 @@ public String lookupServiceProviderUrl(final String catalogUrl, final String ser
 	{
 		String retval = null;
```
Comment (berezovskyi) [inline oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java]: what is this?

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java [hunk 6]
```diff
@@ -672,6 +676,10 @@ else if (this.catalogDomain.equalsIgnoreCase(OSLCConstants.OSLC_AUTO)) {
 
 		try {
 			Response response = this.getResource(rootServicesUrl,OSLCConstants.CT_RDF);
+			if (response.getStatus() != HttpStatus.SC_OK) {
+				logger.error("Cannot read "+rootServicesUrl+" status: "+response.getStatus());
```
Comment (berezovskyi) [inline oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java]: SLF4j supports formatted strings, eg

    logger.error("Cannot read {} status: {}", rootServicesUrl, response.getStatus());

**Comment by berezovskyi** [comment]
Additionally, the build is broken and this needs to be fixed before merging (or in another PR).

**Comment by berezovskyi** [comment]
I made some updates on master to fix the build. @jamsden please pull the latest master and merge the master branch into `jra-rename` (while on the `jra-rename`).

File: oslc4j-client-samples/src/main/java/org/eclipse/lyo/client/oslc/samples/RMSample.java (hunk 3 above)
Comment (jamsden) [reply oslc4j-client-samples/src/main/java/org/eclipse/lyo/client/oslc/samples/RMSample.java]: Fixed

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java (hunk 4 above)
Comment (jamsden) [reply oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java]: changed to return retval (which will be null in this case)

File: oslc-java-samples/pom.xml [hunk 7]
```diff
@@ -11,7 +11,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <wink.version>1.4</wink.version>
-        <lyo.version>2.3.0</lyo.version>
+        <lyo.version>2.4.0</lyo.version>
```
Comment (berezovskyi) [inline oslc-java-samples/pom.xml]: the samples have been moved to https://github.com/OSLC/lyo-samples/tree/master/lyo-client-samples

we need to remove them from this repo

File: oslc-java-samples/src/main/java/org/eclipse/lyo/client/oslc/samples/automation/adapter.properties [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
+serverUrl=https://ce4iot.rtp.raleigh.ibm.com:9443/qm
 # Username for logging into the Automation Provider
-username=TestJazzAdmin1
+username=devonce
 # Password for logging into the Automation Provider
-password=password
+password=devonce
 # Project area in the Automation Provider that the Adapter should be created in 
-projectArea=Quality Manager
+projectArea=JKE Banking (Quality management)
```
Comment (berezovskyi) [inline oslc-java-samples/src/main/java/org/eclipse/lyo/client/oslc/samples/automation/adapter.properties]: haha, credentials leaking again :)

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/exception/FixMessageFormat.java [hunk 9]
```diff
@@ -16,7 +16,7 @@
  *     Chris Peters         - initial API and implementation
  *     Gianluca Bernardini  - initial API and implementation
  *******************************************************************************/
-package org.eclipse.lyo.client.exception;
+package org.eclipse.lyo.oslc4j.client.exception;
 
 public class FixMessageFormat {
```
Comment (berezovskyi) [inline oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/exception/FixMessageFormat.java]: This is too much duplication in the new client. Should not be like that!

**Comment by berezovskyi** [comment]
thanks for updating package names, please see my other comments

**Comment by berezovskyi** [comment]
This PR has to be updated given the sample code is now moved to https://github.com/OSLC/lyo-samples

**Comment by berezovskyi** [comment]
Jersey version will be downgraded as per https://github.com/OSLC/lyo-adaptor-sample-modelling/commit/991d2e3a7955d07259f18bce1399a76d2f5a96e6

**Comment by berezovskyi** [comment]
Just merged latest master into this and resolved merge conflicts.

**Comment by berezovskyi** [comment]
We will deal with my other comments in #40 and other PRs.

**Pull request merged**